management/server/dns.go

```diff
@@ -133,26 +133,31 @@ func (am *DefaultAccountManager) SaveDNSSettings(ctx context.Context, accountID
 	oldSettings := account.DNSSettings.Copy()
 	account.DNSSettings = dnsSettingsToSave.Copy()
 
+	addedGroups := difference(dnsSettingsToSave.DisabledManagementGroups, oldSettings.DisabledManagementGroups)
+	removedGroups := difference(oldSettings.DisabledManagementGroups, dnsSettingsToSave.DisabledManagementGroups)
+
 	account.Network.IncSerial()
 	if err = am.Store.SaveAccount(ctx, account); err != nil {
 		return err
 	}
 
-	addedGroups := difference(dnsSettingsToSave.DisabledManagementGroups, oldSettings.DisabledManagementGroups)
 	for _, id := range addedGroups {
 		group := account.GetGroup(id)
 		meta := map[string]any{"group": group.Name, "group_id": group.ID}
 		am.StoreEvent(ctx, userID, accountID, accountID, activity.GroupAddedToDisabledManagementGroups, meta)
 	}
 
-	removedGroups := difference(oldSettings.DisabledManagementGroups, dnsSettingsToSave.DisabledManagementGroups)
 	for _, id := range removedGroups {
 		group := account.GetGroup(id)
 		meta := map[string]any{"group": group.Name, "group_id": group.ID}
 		am.StoreEvent(ctx, userID, accountID, accountID, activity.GroupRemovedFromDisabledManagementGroups, meta)
 	}
 
-	am.updateAccountPeers(ctx, account)
+	updateAccountPeers := (areGroupChangesAffectPeers(account, addedGroups) && anyGroupHasPeers(account, addedGroups)) ||
+		areGroupChangesAffectPeers(account, removedGroups) && anyGroupHasPeers(account, removedGroups)
+	if updateAccountPeers {
+		am.updateAccountPeers(ctx, account)
+	}
```

Comment:
In my opinion this check is wrong. The check for `anyGroupHasPeers(account, addedGroups) || anyGroupHasPeers(account, removedGroups)` is correct but we don't need `areGroupChangesAffectPeers`. This will cause wrong updates. E.g. I have removed a group from the nameserver so the add case will fasle, the `anyGroupHasPeers(account, removedGroups)` will be true but `areGroupChangesAffectPeers(account, removedGroups)` will also false as the dependency has been removed with this change. So long story short, we don''t need to be so complex, `anyGroupHasPeers` for added and removed groups should already be enough

Reply:
Ah I was in the wrong method. The case I described is not applicable here. Still `isGroupLinkedToDns(account.NameServerGroups, groupID)` shoudl be enough